src/EditorFeatures/Core/Implementation/IntelliSense/IDocumentProvider.cs

```diff
@@ -24,10 +22,10 @@ public DocumentProvider(IThreadingContext threadingContext)
         {
         }
 
-        public Document GetDocument(ITextSnapshot snapshot, CancellationToken cancellationToken)
+        public ValueTask<Document?> GetDocumentAsync(ITextSnapshot snapshot, CancellationToken cancellationToken)
```

Comment:
terrifying that thsi doesn't state taht it's getting frozen partial.  i know that's what it was before, but i am genuinely baffled by this.  also, do we even need this type?  any caller can just do the code below right?